app/controllers/api_keys_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ApiKeysController < ApplicationController
+  before_action :redirect_to_root, unless: :signed_in?
+  before_action :verify_authorization, only: :destroy
+
+  def index
+    @api_keys = current_user.api_keys
+    redirect_to new_profile_api_key_path if @api_keys.empty?
+  end
+
+  def new
+    @api_key = current_user.api_keys.build
+  end
+
+  def create
+    @key = SecureRandom.hex(16)
```

Comment:
Consider increasing the complexity of the keys:

method|characters|total available characters
--------|-----------|------------------------------
`SecureRandom.hex` (current)|`0-9`, `a-f`|`16`
`SecureRandom.alphanumeric `|`A-Z`, `a-z`, `0-9`|`62`
`SecureRandom.urlsafe_base64 `|`A-Z`, `a-z`, `0-9`, `-`,`_`|`64`

Reply:
hex, alphanumeric are just a representation of 16 random bytes and doesn't affect complexity of guessing the key.
```
SecureRandom.hex(16)
 => "6b374974e76ad2849d894210d1bcfc09"
SecureRandom.alphanumeric(16)
 => "taEC0njBvhZlgqX7"
``` 

We could increase the number of bytes but I would argue 16 bytes is safe size to use when we are expiring keys in 6 months.